File: typescript/resistor-color-duo/resistor-color-duo.ts
export class ResistorColor {
  constructor(
    private colors: string[]
    ) {
    if (colors.length < 2) {
      throw(Error('At least two colors need to be present'))
    }
  }

  value = (): number => {
    let tens = RESISTOR_COLORS.indexOf(this.colors[0]);
    let units = RESISTOR_COLORS.indexOf(this.colors[1]);

    return (tens * 10 + units)
  }
}


const RESISTOR_COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white'
]

File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  def self.value(colors)
    colors
      .first(2)
      .map(&:capitalize)
      .map { |c| COLOR_MAP.index(c) }
      .join
      .to_i
  end

  COLOR_MAP = %w[
    Black Brown Red Orange Yellow Green Blue Violet Grey White
  ].freeze
end